samples/cpp/text_generation/encrypted_model_causal_lm.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/llm_pipeline.hpp"
+#include <fstream>
+
+std::pair<std::string, ov::Tensor> decrypt_model(const std::string& model_path, const std::string& weights_path) {
+        std::ifstream model_file(model_path);
+    std::ifstream weights_file(weights_path, std::ios::binary);
+    if (!model_file.is_open() || !weights_file.is_open()) {
+        throw std::runtime_error("Cannot open model or weights file");
+    }
+
+    // User can add file decryption of model_file and weights_file in memory here.
+
+    std::string model_str((std::istreambuf_iterator<char>(model_file)), std::istreambuf_iterator<char>());
+    std::vector<char> weights_buffer((std::istreambuf_iterator<char>(weights_file)), std::istreambuf_iterator<char>());
+    auto weights_tensor = ov::Tensor(ov::element::u8, {weights_buffer.size()}, weights_buffer.data());
+    return {model_str, weights_tensor};
+}
+
+ov::genai::Tokenizer decrypt_tokenizer(const std::string& models_path) {
+    std::string tok_model_path = models_path + "/openvino_tokenizer.xml";
+    std::string tok_weights_path = models_path + "/openvino_tokenizer.bin";
+    auto [tok_model_str, tok_weights_tensor] = decrypt_model(tok_model_path, tok_weights_path);
+
+    std::string detok_model_path = models_path + "/openvino_detokenizer.xml";
+    std::string detok_weights_path = models_path + "/openvino_detokenizer.bin";
+    auto [detok_model_str, detok_weights_tensor] = decrypt_model(tok_model_path, tok_weights_path);
+
+    return ov::genai::Tokenizer(tok_model_str, tok_weights_tensor, detok_model_str, detok_weights_tensor);
+}
+
+int main(int argc, char* argv[]) try {
+    if (3 > argc)
+        throw std::runtime_error(std::string{"Usage: "} + argv[0] + " <MODEL_DIR> \"<PROMPT>\"");
+
+    std::string device = "CPU";  // GPU, NPU can be used as well
+    std::string models_path = argv[1];
+    std::string prompt = argv[2];
+
+    auto [model_str, model_weights] = decrypt_model(models_path + "/openvino_model.xml", models_path + "/openvino_model.bin");
+    ov::genai::Tokenizer tokenizer = decrypt_tokenizer(models_path);
+    
+    // NPU reads some properties from the config file, but when LLMPipeline is initialized 
+    // from the model_str and weights_tensor, there not such folder. Therefore, we need to
+    // pass these properties manually.
+    // This is necessary only for NPU, for other plugins can be ommited.
+    ov::AnyMap model_descr_properties = {{"name_or_path", "meta-llama/Llama-2-7b-chat-hf"}, 
+                                         {"type", "llama"}, 
+                                         {"num_key_value_heads", 32}};
```

Comment:
maybe we can avoid demonstrating such NPU hacks in samples? 

I hope that @dmatveev and @TolyaTalamanov can come up with more generic approach which does not require reading this information from `config.json`. E.g. `num_key_value_heads` can be extracted from model itself
https://github.com/openvinotoolkit/openvino.genai/blob/e2fa0d002117bbfe6d9a6a4cc413e583c4455131/src/cpp/src/utils/paged_attention_transformations.cpp#L23-L25